### PR TITLE
Add option to allow cross-site requests

### PR DESCRIPTION
**Pull request by zonque**
This allows users to add CORS headers to responses of the server.

### CODE REVIEW

**Comment by chclaus** [comment]
Hi @zonque,

thanks a lot for this contribution! I will have a look asap! 

Best,
Christian

File: Readme.md
Comment (chclaus) [inline Readme.md:101]: Could you please double-quote the wildcard?

```
cors:
    origin: "*"
```

File: app/config.go
Comment (chclaus) [inline app/config.go:49]: I think it would be nice and complete, if you could provide configuration options for the fields CORS fields:

- Allow-Header
- Allow Methods

File: cmd/dave/main.go
Comment (chclaus) [inline cmd/dave/main.go:81]: Here you could insert the other configurable config.Cors properties if introduced before.

**Review by chclaus — CHANGES_REQUESTED** [review]
Hi @zonque,

the code works exactly as expected. Also thanks for thinking of the documentation.

I made a few comments that I would like to see implemented before hitting the merge button. ;)

I would also like to see the new additional properties in examples/config-sample.yaml. Commented out of course. ;)

File: Readme.md
Comment (zonque) [reply Readme.md]: Done.

File: app/config.go
Comment (zonque) [reply app/config.go:49]: I originally thought about that, but then decided not to. The reason is that I don't see any benefit for the user really, as one needs alls the methods and headers involved in WebDAV to ensure functionalty You either want to allow CORS in your setup or you don't, and selecting the hosts to enable this for should suffice. Only allowing specific headers or methods will break the WebDAV operation. WDYT?

File: cmd/dave/main.go
Comment (zonque) [reply cmd/dave/main.go:81]: > I would also like to see the new additional properties in examples/config-sample.yaml. Commented out of course. ;)

Done.

File: app/config.go
Comment (chclaus) [reply app/config.go:49]: Ok, you gut me

File: cmd/dave/main.go
Comment (chclaus) [reply cmd/dave/main.go:81]: Thanks :)

**Comment by chclaus** [comment]
Thanks a lot @zonque,

I'll try to make a new release these days.

Best,
Christian

**Pull request merged**
